src/modules/launcher/Plugins/Microsoft.Plugin.WindowWalker/Components/ResultHelper.cs

```diff
@@ -0,0 +1,160 @@
+                    SubTitle = GetSubtitle(x.Result),
+                    ContextData = x.Result,
+                    Action = c =>
+                    {
+                        x.Result.SwitchToWindow();
+                        return true;
+                    },
+
+                    // For debugging you can set the second parameter to true to see more informations.
+                    ToolTipData = GetToolTip(x.Result, false),
+                });
+            }
+
+            if (addExplorerInfo && isKeywordSearch && !WindowWalkerSettings.Instance.HideExplorerSettingInfo)
+            {
+                resultsList.Add(GetExplorerInfoResult(infoIcon));
+            }
+
+            return resultsList;
+        }
+
+        /// <summary>
+        /// Returns the subtitle for a result
+        /// </summary>
+        /// <param name="window">The window properties of the result</param>
+        /// <returns>String with the subtitle</returns>
+        private static string GetSubtitle(Window window)
+        {
+            if (window == null || !(window is Window))
+            {
+                return string.Empty;
+            }
+
+            string subtitleText = Resources.wox_plugin_windowwalker_Running + ": " + window.Process.Name;
+
+            if (WindowWalkerSettings.Instance.SubtitleShowPid)
+            {
+                subtitleText += $" ({window.Process.ProcessID})";
+            }
+
+            if (WindowWalkerSettings.Instance.SubtitleShowDesktopName && Main.VirtualDesktopHelperInstance.GetDesktopCount() > 1)
+            {
+                subtitleText += $" - {Resources.wox_plugin_windowwalker_Desktop}: {window.Desktop.Name}";
+            }
+
+            return subtitleText;
+        }
+
+        /// <summary>
+        /// Returns the tool tip for a result
+        /// </summary>
+        /// <param name="window">The window properties of the result</param>
+        /// <param name="debugToolTip">Value indicating if a detailed debug tooltip should be returned</param>
+        /// <returns>Tooltip for the result or null of failure</returns>
+        private static ToolTipData GetToolTip(Window window, bool debugToolTip)
+        {
+            if (window == null || !(window is Window))
+            {
+                return null;
+            }
+
+            if (!debugToolTip)
+            {
+                string text = $"{Resources.wox_plugin_windowwalker_Process}: {window.Process.Name}";
+                text += $"\n{Resources.wox_plugin_windowwalker_ProcessId}: {window.Process.ProcessID}";
+
+                if (Main.VirtualDesktopHelperInstance.GetDesktopCount() > 1)
+                {
+                    text += $"\n{Resources.wox_plugin_windowwalker_Desktop}: {window.Desktop.Name}";
+
+                    if (!window.Desktop.IsAllDesktopsView)
+                    {
+                        text += $" ({Resources.wox_plugin_windowwalker_Number} {window.Desktop.Number})";
```

Comment:
Note to me: If reviewers think this is unclear we can switch to `{0} Desktop` like `1 Desktop`, `2 Desktop`.